src/hotspot/share/runtime/vmStructs.hpp

```diff
@@ -187,19 +188,23 @@ class VMStructs {
 
 #ifdef ASSERT
 
+// Converts any type T to a reference type.
+template<typename T>
+std::add_rvalue_reference_t<T> declval() noexcept;
```

Comment:
I think I'd prefer this was in globalDefinitions.hpp rather than squirreled away in this obscure location.  But maybe that can wait until someone else needs it.  (And maybe by then we can use std::declval :) )

Reply:
I tried that, but it needed <type_traits> in globalDefinitions.hpp, so I backed off. How's this?